### ISSUE TITLE
Get branch coverage with clang-13

### ISSUE THREAD

**Issue by chfast** [ethereum/evmone#398]
First of all, when upgrading CI to clang-13 to do coverage report, the `llvm-profdata` reports malformed data. So we need to make it work with clang-13 first. Try reproducing the issue locally with clang-13 toolchain.

The CI steps:  https://github.com/ethereum/evmone/blob/dca15dfd3a7a998b38821061e329d54df41484e6/circle.yml#L335-L357
Clang documentation: https://clang.llvm.org/docs/SourceBasedCodeCoverage.html

Additionally, clang-13 code coverage has some new features like branch coverage, so try to present these on a report.

**Comment by chfast** [comment]
This has been resolved in EVMC. https://github.com/ethereum/evmc/pull/619/files#diff-2f3053ad458fe5dcd4194e9e8cccfca390e99bf1b046029a0b2cbb21428eeafeR91

**Comment by rodiazet** [comment]
It looks like this solution doesn't work. According to https://groups.google.com/g/llvm-dev/c/QQ7GKZLN5dE?pli=1 there is an issue with `llvm-profdata` for `clang 13.0.0` which has been solved and delivered with `clang 13.0.1` (two days ago :)). I tested it using prepared docker image with new `clang` environment available on `imappdocker/cpp-build-env:17-clang-13` and everything works fine. So it seems to be a matter of updating `ethereum/cpp-build-env:17-clang-13` docker image.
